src/Microsoft.AspNetCore.Authentication.MicrosoftAccount/MicrosoftAccountHelper.cs

```diff
@@ -34,8 +35,17 @@ public static string GetName(JObject user)
             {
                 throw new ArgumentNullException(nameof(user));
             }
+            
+            //displayName is required for Azure AD accounts, but not for Microsoft accounts.
+            //givenName & surename is required for Microsoft accounts, but not for Azure AD accounts
+            //In case of microsoft accounts displayName is null => fallback to givenName + surename
+            var name = user.Value<string>("displayName");
+            if (string.IsNullOrEmpty(name))
+            {
+                name = string.Format("{0} {1}", GetFirstName(user), GetLastName(user));
```

Comment:
Remove this. We don't want to transform the claims we receive, just pass them through. It's up to apps to make these kind of transformation decisions, especially for something like this can can be culture sensitive.